connectors/sources/sharepoint_online.py

```diff
@@ -793,6 +793,14 @@ def get_default_configuration(cls):
                 "type": "bool",
                 "value": False,
             },
+            "use_document_level_security": {
+                "display": "toggle",
```

Comment:
Good point, would also apply to the extraction service I guess.

Reply:
@wangch079 @timgrein we could expand `depends_on` for that use-case. It isn't philosophically tied to other configurable fields, that's just what it's only used for right now. I could see `depends_on: [{ field: "feature_flag", value: "heyimafeatureflag" }]` existing, for example.

The only possible issue I could see if we added that would be with validations. AFAIK we don't have access to the license from within connectors-python (and maybe not feature flags either?), so we would attempt to validate them in the backend at run-time, but the user may not be able to edit them if they have the wrong license.

Reply:
>AFAIK we don't have access to the license from within connectors-python (and maybe not feature flags either?)

We have both. Feature flags are available in the connector document. And we are already checking license with [Get license API](https://www.elastic.co/guide/en/elasticsearch/reference/current/get-license.html).

I believe this will require FE changes. Hi @efegurkan are you planning to have a special checking for SPO (check license and feature flags), or do you think we can integrate this into the scope of Rich Configurable Fields?